tests/conftest.py

```diff
@@ -139,7 +139,23 @@ def tmpdir(request, tmpdir):
     deleting the temporary directories at the end of each test case.
     """
     assert tmpdir.isdir()
-    yield Path(str(tmpdir))
+    # On some *nix systems bound to AD domains, the $USER environment is of the
+    # form `DOMAIN\username`, which generates a tmpdir of the form:
+    #
+    # /tmp/pytest-of-DOMAN\username/pytest-XX/test_nameX
+    #
+    # When passed into pip commands, these paths get translated into:
+    #
+    # /tmp/pytest-of-DOMANusername/pytest-XX/test_nameX
```

Comment:
Why does this happen?

Reply:
Hi @pradyunsg 

> Thanks for filing this PR @lebedov.

I'm @leorochael, not @lebedov.

> I'm not sure what this PR is intending to achieve.

This PR fixes tests that fail on my machine because of the interaction of the `tmpdir` fixture in `conftest.py` and the `$USER` variable which, on my machine, contains a backslash (e.g. `DOMAIN\username`).

This doesn't happen in normal `pip` operation since `pip` doesn't try to derive a path from the `$USER` variable (any user-related pip operation relies on `$HOME` instead, which doesn't have a backslash).

Only the `tmpdir` fixture tries to derive a pathname from the `$USER` variable, and this pathname is then abused in different ways by different fixtures that don't expect slashes mixed with backslashes, like the `script` fixture. and the `script.pip()` command.

> If the intent is to fix the improper handling of the path, I am certain that changes need to happen somewhere in the `pip` package and not inside tests. Further, this doesn't look like an appropriate way to even test a bugfix since it's not reproducing the conditions under which the failures occur.

That's the thing... there is no improper handling of the path in `pip`. There's only improper handling of the `$USER` variable by test fixtures.

I haven't figured out a way to fake `$USER` **before** a fixture is run during a test, and even if I did, I'd then need to write a test to test fixtures, which seems overkill.

Reply:
> I'm @leorochael, not @lebedov.

Oh heck. My bad. Sorry! 😓 

> This PR fixes tests that fail on my machine because of the interaction of the tmpdir fixture in conftest.py and the $USER variable which, on my machine, contains a backslash (e.g. DOMAIN\username).

Ah, this is useful context that I wasn't able to understand by merely reading the PR.

Reply:
> Ah, this is useful context that I wasn't able to understand by merely reading the PR.

I just realised the PR description was empty. Fixed it now.